Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostChannel/NvHostChannelDeviceFile.cs

```diff
@@ -0,0 +1,352 @@
+            {
+                NvMapHandle map = NvMapDeviceFile.GetMapFromHandle(_owner, commandBufferEntry.MapHandle);
+
+                if (map == null)
+                {
+                    Logger.PrintWarning(LogClass.ServiceNv, $"Invalid handle 0x{commandBufferEntry.MapHandle:x8}!");
+
+                    return NvInternalResult.InvalidInput;
+                }
+
+                lock (map)
+                {
+                    if (map.DmaMapAddress == 0)
+                    {
+                        map.DmaMapAddress = vmm.MapLow(map.Address, map.Size);
+                    }
+
+                    commandBufferEntry.MapAddress = (int)map.DmaMapAddress;
+                }
+            }
+
+            return NvInternalResult.Success;
+        }
+
+        private NvInternalResult UnmapCommandBuffer(Span<byte> arguments)
+        {
+            int                       headerSize           = Unsafe.SizeOf<MapCommandBufferArguments>();
+            MapCommandBufferArguments commandBufferHeader  = MemoryMarshal.Cast<byte, MapCommandBufferArguments>(arguments)[0];
+            Span<CommandBufferHandle> commandBufferEntries = MemoryMarshal.Cast<byte, CommandBufferHandle>(arguments.Slice(headerSize)).Slice(0, commandBufferHeader.NumEntries);
+            NvGpuVmm                  vmm                  = NvHostAsGpuDeviceFile.GetAddressSpaceContext(_owner).Vmm;
+
+            // TODO
+
+            foreach (ref CommandBufferHandle commandBufferEntry in commandBufferEntries)
+            {
+                NvMapHandle map = NvMapDeviceFile.GetMapFromHandle(_owner, commandBufferEntry.MapHandle);
+
+                if (map == null)
+                {
+                    Logger.PrintWarning(LogClass.ServiceNv, $"Invalid handle 0x{commandBufferEntry.MapHandle:x8}!");
+
+                    return NvInternalResult.InvalidInput;
+                }
+
+                lock (map)
+                {
+                    if (map.DmaMapAddress != 0)
+                    {
+                        vmm.Free(map.DmaMapAddress, map.Size);
+
+                        map.DmaMapAddress = 0;
+                    }
+                }
+            }
+
+            return NvInternalResult.Success;
+        }
+
+        private NvInternalResult SetNvMapFd(ref int nvMapFd)
+        {
+            Logger.PrintStub(LogClass.ServiceNv);
+
+            return NvInternalResult.Success;
+        }
+
+        private NvInternalResult SetTimeout(ref uint timeout)
+        {
+            _timeout = timeout;
+
+            Logger.PrintStub(LogClass.ServiceNv);
+
+            return NvInternalResult.Success;
+        }
+
+        private NvInternalResult SubmitGpfifo(Span<byte> arguments)
+        {
+            int                   headerSize                = Unsafe.SizeOf<SubmitGpfifoArguments>();
+            SubmitGpfifoArguments gpfifoSubmissionHeader    = MemoryMarshal.Cast<byte, SubmitGpfifoArguments>(arguments)[0];
```

Comment:
```suggestion
            SubmitGpfifoArguments gpfifoSubmissionHeader = MemoryMarshal.Cast<byte, SubmitGpfifoArguments>(arguments)[0];
```
and same for other vars